srt.py

```diff
@@ -18,11 +18,14 @@
 # accept it, so we do too.
 RGX_TIMESTAMP_MAGNITUDE_DELIM = r"[,.:，．。：]"
 RGX_TIMESTAMP = RGX_TIMESTAMP_MAGNITUDE_DELIM.join([r"\d+"] * 4)
+RGX_TIMESTAMP_PARSEABLE = RGX_TIMESTAMP_MAGNITUDE_DELIM.join(
+        [r"(\d+)", r"(\d{2})", r"(\d{2})", r"(\d{1,3})"])
 RGX_INDEX = r"\d+"
 RGX_PROPRIETARY = r"[^\r\n]*"
 RGX_CONTENT = r".*?"
 RGX_POSSIBLE_CRLF = r"\r?\n"
 
+TS_REGEX = re.compile(RGX_TIMESTAMP_PARSEABLE)
```

Comment:
I think you need to anchor the end on this one, otherwise:

```
In [7]: TS_REGEX.match("01:01:02,003")                                                                                                                        
Out[7]: <re.Match object; span=(0, 12), match='01:01:02,003'>

In [8]: TS_REGEX.match("01:01:02,0034")                                                                                                                       
Out[8]: <re.Match object; span=(0, 12), match='01:01:02,003'>
```

For the second case, we should get an SRTParseError I suppose

Reply:
@cdown I think by changing the ms regex from "\d{1,3}" to "\d+" we'll obviate the need for an anchor since the matching should be greedy after we allow arbitrary-length timestamp fields.

Reply:
Eh, I went ahead and surrounded the ts regex by ^ and $ because it can't hurt anything.